foreign-object/src/Foreign/Object.ss

```diff
@@ -0,0 +1,159 @@
+;; -*- mode: scheme -*-
+
+(library (Foreign.Object foreign)
+  (export _copyST
+          empty
+          runST
+          _fmapObject
+          _mapWithKey
+          _foldM
+          _foldSCObject
+          all
+          size
+          _lookup
+          _lookupST
+          fromHomogeneousImpl
+          toArrayWithKey
+          keys)
+  (import (only (rnrs base) define lambda if let let* let-values begin)
+          (prefix (chezscheme) scm:)
```

Comment:
I've imported this unqualified, to make the code a bit more terse, but it's fine if you want to qualify.